Warnings when compiling with GCC 14.3

Compiling Palace with a recent GCC leads to some warnings that we can easily fix.

```cpp
/data/home/gbozzola/palace/palace/models/postoperatorcsv.cpp: In instantiation of ‘std::enable_if_t<(((U == palace::ProblemType::EIGENMODE) || (U == palace::ProblemType::DRIVEN)) || (U == palace::ProblemType::TRANSIENT)), void> palace::PostOperatorCSV<solver_t>::PrintPortVI(const palace::LumpedPortOperator&, const palace::Units&) [with palace::ProblemType U = palace::ProblemType::DRIVEN; palace::ProblemType solver_t = palace::ProblemType::DRIVEN; std::enable_if_t<(((U == palace::ProblemType::EIGENMODE) || (U == palace::ProblemType::DRIVEN)) || (U == palace::ProblemType::TRANSIENT)), void> = void]’:
/data/home/gbozzola/palace/palace/models/postoperatorcsv.cpp:1296:16:   required from ‘void palace::PostOperatorCSV<solver_t>::PrintAllCSVData(const palace::PostOperator<solver_t>&, const palace::Measurement&, double, int) [with palace::ProblemType solver_t = palace::ProblemType::DRIVEN]’
 1296 |     PrintPortVI(post_op.fem_op->GetLumpedPortOp(), post_op.units);
      |     ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/home/gbozzola/palace/palace/models/postoperatorcsv.cpp:1373:16:   required from here
 1373 | template class PostOperatorCSV<ProblemType::DRIVEN>;
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/home/gbozzola/palace/palace/models/postoperatorcsv.cpp:975:27: warning: comparison of integer expressions of different signedness: ‘const int’ and ‘std::size_t’ {aka ‘long unsigned int’} [-Wsign-compare]
  975 |       if (data.excitation == m_ex_idx)
      |           ~~~~~~~~~~~~~~~~^~~~~~~~~~~
[ 92%] Building CXX object CMakeFiles/libpalace.dir/utils/meshio.cpp.o
[ 95%] Building CXX object CMakeFiles/libpalace.dir/utils/omp.cpp.o
/data/home/gbozzola/palace/palace/models/postoperatorcsv.cpp: In instantiation of ‘std::enable_if_t<(((U == palace::ProblemType::EIGENMODE) || (U == palace::ProblemType::DRIVEN)) || (U == palace::ProblemType::TRANSIENT)), void> palace::PostOperatorCSV<solver_t>::PrintPortVI(const palace::LumpedPortOperator&, const palace::Units&) [with palace::ProblemType U = palace::ProblemType::TRANSIENT; palace::ProblemType solver_t = palace::ProblemType::TRANSIENT; std::enable_if_t<(((U == palace::ProblemType::EIGENMODE) || (U == palace::ProblemType::DRIVEN)) || (U == palace::ProblemType::TRANSIENT)), void> = void]’:
/data/home/gbozzola/palace/palace/models/postoperatorcsv.cpp:1296:16:   required from ‘void palace::PostOperatorCSV<solver_t>::PrintAllCSVData(const palace::PostOperator<solver_t>&, const palace::Measurement&, double, int) [with palace::ProblemType solver_t = palace::ProblemType::TRANSIENT]’
 1296 |     PrintPortVI(post_op.fem_op->GetLumpedPortOp(), post_op.units);
      |     ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/home/gbozzola/palace/palace/models/postoperatorcsv.cpp:1377:16:   required from here
 1377 | template class PostOperatorCSV<ProblemType::TRANSIENT>;
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/home/gbozzola/palace/palace/models/postoperatorcsv.cpp:975:27: warning: comparison of integer expressions of different signedness: ‘const int’ and ‘std::size_t’ {aka ‘long unsigned int’} [-Wsign-compare]
  975 |       if (data.excitation == m_ex_idx)
      |           ~~~~~~~~~~~~~~~~^~~~~~~~~~~
[ 95%] Building CXX object CMakeFiles/libpalace.dir/utils/tablecsv.cpp.o
/data/home/gbozzola/palace/palace/linalg/nleps.cpp: In function ‘palace::ComplexVector palace::{anonymous}::MatVecMult(const std::vector<palace::ComplexVector>&, const Eigen::VectorXcd&
’:
/data/home/gbozzola/palace/palace/linalg/nleps.cpp:340:21: warning: comparison of integer expressions of different signedness: ‘int’ and ‘const size_t’ {aka ‘const long unsigned int’} [-Wsign-compare]
  340 |   for (int j = 0; j < k; j++)
      |                   ~~^~~
In file included from /data/home/gbozzola/palace/c7g_coverage/include/mfem/mfem.hpp:17,
                 from /data/home/gbozzola/palace/c7g_coverage/include/mfem.hpp:2,
                 from /data/home/gbozzola/palace/palace/utils/configfile.cpp:11:
/data/home/gbozzola/palace/palace/utils/configfile.cpp: In lambda function:
/data/home/gbozzola/palace/palace/utils/configfile.cpp:1928:35: warning: suggest parentheses around comparison in operand of ‘^’ [-Wparentheses]
 1928 |               MFEM_VERIFY(delta_f > 0 ^ n_sample > 0,
      |                           ~~~~~~~~^~~
In file included from /opt/gcc/include/c++/14.3.0/bits/stl_algo.h:61,
                 from /opt/gcc/include/c++/14.3.0/algorithm:61,
                 from /data/home/gbozzola/palace/palace/utils/configfile.cpp:6:
/opt/gcc/include/c++/14.3.0/bits/stl_heap.h: In function ‘void std::__adjust_heap(_RandomAccessIterator, _Distance, _Distance, _Tp, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<pair<double, double>*, vector<pair<double, double> > >; _Distance = long int; _Tp = pair<double, double>; _Compare = __gnu_cxx::__ops::_Iter_less_iter]’:
/opt/gcc/include/c++/14.3.0/bits/stl_heap.h:224:5: note: parameter passing for argument of type ‘std::pair<double, double>’ when C++17 is enabled changed to match C++14 in GCC 10.1
  224 |     __adjust_heap(_RandomAccessIterator __first, _Distance __holeIndex,
      |     ^~~~~~~~~~~~~
```

--- a/palace/linalg/nleps.cpp
+++ b/palace/linalg/nleps.cpp
@@ -329,9 +329,11 @@ namespace
 // size k, returning a ComplexVector of size n.
 ComplexVector MatVecMult(const std::vector<ComplexVector> &X, const Eigen::VectorXcd &y)
 {
-  MFEM_ASSERT(X.size() == y.size(), "Mismatch in dimension of input vectors!");
-  const size_t k = X.size();
-  const size_t n = X[0].Size();
+  // Cast to avoid compiler warnings about types.
+  MFEM_ASSERT(static_cast<Eigen::Index>(X.size()) == y.size(),
+              "Mismatch in dimension of input vectors!");
+  const int k = X.size();
+  const int n = X[0].Size();
   const bool use_dev = X[0].UseDevice();
   ComplexVector z;
   z.SetSize(n);

--- a/palace/models/postoperator.hpp
+++ b/palace/models/postoperator.hpp
@@ -133,7 +133,7 @@ protected:
   bool ShouldWriteFields() const { return AnyOutputFormats() && AnythingToSave(); }
 
   // Whether any fields should be written for this step.
-  bool ShouldWriteFields(std::size_t step) const
+  bool ShouldWriteFields(int step) const
   {
     return AnyOutputFormats() &&
            ((output_delta_post > 0 && step % output_delta_post == 0) ||
@@ -147,7 +147,7 @@ protected:
   {
     return enable_paraview_output && AnythingToSave();
   }
-  bool ShouldWriteParaviewFields(std::size_t step) const
+  bool ShouldWriteParaviewFields(int step) const
   {
     return enable_paraview_output && ShouldWriteFields(step);
   }
@@ -155,7 +155,7 @@ protected:
   {
     return enable_gridfunction_output && AnythingToSave();
   }
-  bool ShouldWriteGridFunctionFields(std::size_t step) const
+  bool ShouldWriteGridFunctionFields(int step) const
   {
     return enable_gridfunction_output && ShouldWriteFields(step);
   }

--- a/palace/models/postoperatorcsv.cpp
+++ b/palace/models/postoperatorcsv.cpp
@@ -976,7 +976,8 @@ auto PostOperatorCSV<solver_t>::PrintPortVI(const LumpedPortOperator &lumped_por
   {
     for (const auto &[idx, data] : lumped_port_op)
     {
-      if (data.excitation == m_ex_idx)
+      // Cast to avoid compiler warnings about types.
+      if (static_cast<std::size_t>(data.excitation) == m_ex_idx)
       {
         auto Jcoeff = measurement_cache.Jcoeff_excitation;
         double V_inc = data.GetExcitationVoltage() * Jcoeff;

--- a/palace/utils/configfile.cpp
+++ b/palace/utils/configfile.cpp
@@ -1584,7 +1584,8 @@ void FarFieldPostData::SetUp(json &postpro)
       }
 
       if (nsample > 2)
-        MFEM_ASSERT(thetaphis.size() == nsample,
+        // Cast to avoid compiler warnings about types.
+        MFEM_ASSERT(static_cast<int>(thetaphis.size()) == nsample,
                     "Sampled number of points is not NSample!");
     }
   }
@@ -1937,7 +1938,7 @@ void DrivenSolverData::SetUp(json &solver)
               auto max_f = r.at("MaxFreq");
               auto delta_f = r.value("FreqStep", 0.0);
               auto n_sample = r.value("NSample", 0);
-              MFEM_VERIFY(delta_f > 0 ^ n_sample > 0,
+              MFEM_VERIFY((delta_f > 0) ^ (n_sample > 0),
                           "Only one of \"FreqStep\" or \"NSample\" can be specified for "
                           "\"Type\": \"Linear\"!");
               if (delta_f > 0)

